roles/k3s/master/tasks/metallb.yml

```diff
@@ -17,11 +17,23 @@
     mode: 0644
   when: ansible_hostname == hostvars[groups['master'][0]]['ansible_hostname']
 
-- name: Copy metallb manifest to first master
-  template:
-    src: "metallb.crds.j2"
+- name: Download metallb manifest to first master
+  ansible.builtin.get_url:
+    url: "https://raw.githubusercontent.com/metallb/metallb/{{ metal_lb_controller_tag_version }}/config/manifests/metallb-native.yaml"  # noqa yaml[line-length]
     dest: "/var/lib/rancher/k3s/server/manifests/metallb-crds.yaml"
     owner: root
     group: root
     mode: 0644
   when: ansible_hostname == hostvars[groups['master'][0]]['ansible_hostname']
+
+- name: Set image versions for metallb manifest
+  ansible.builtin.replace:
+    path: "/var/lib/rancher/k3s/server/manifests/metallb-crds.yaml"
+    regexp: "{{ item.change | ansible.builtin.regex_escape }}"
+    replace: "{{ item.to }}"
+  with_items:
+    - change: "metallb/speaker:{{ metal_lb_controller_tag_version }}"
+      to: "metallb/speaker:{{ metal_lb_speaker_tag_version }}"
+  loop_control:
+    label: "{{ item.change }} => {{ item.to }}"
+  when: ansible_hostname == hostvars[groups['master'][0]]['ansible_hostname']
```

Comment:
@timothystewart6 With the original implementation of the downloaded manifests, the controller and speaker tags are seperated into two different variables. Since these versions always seem to match in the original manifests, is it necessary to split these into two variables? If not, the regex here would be totally unneccesary.

Reply:
You're right, they should be lockstep but unfortunately that's not how it started and I don't want to break this for anyone who updates to the latest playbook.  When I originally wrote it I thought we might end up in the opposite scenario where someone needs to change the version one and not the other.   I decided since that they were different containers with different build processes it was best to separate them just in case one they they are no longer lockstep upstream.